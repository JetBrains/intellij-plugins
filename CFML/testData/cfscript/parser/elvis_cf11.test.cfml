component {
  var securityStruct = {};
  securityNumber = securityStruct["Joe"] ?: -1;
}