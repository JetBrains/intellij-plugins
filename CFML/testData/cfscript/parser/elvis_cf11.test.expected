CfmlFile:elvis_cf11.cfml
  COMPONENT_DEFINITION
    PsiElement(COMPONENT_KEYWORD)('component')
    PsiElement(L_CURLYBRACKET)('{')
    Assignment
      PsiElement(VAR_KEYWORD)('var')
      ReferenceExpression
        PsiElement(IDENTIFIER)('securityStruct')
      PsiElement(ASSIGN)('=')
      PsiElement(L_CURLYBRACKET)('{')
      PsiElement(R_CURLYBRACKET)('}')
    PsiElement(SEMICOLON)(';')
    Assignment
      ReferenceExpression
        PsiElement(IDENTIFIER)('securityNumber')
      PsiElement(ASSIGN)('=')
      BinaryExpression
        None
          ReferenceExpression
            PsiElement(IDENTIFIER)('securityStruct')
          PsiElement(L_SQUAREBRACKET)('[')
          StringLiteral
            PsiElement(DOUBLE_QUOTE)('"')
            PsiElement(STRING_TEXT)('Joe')
            PsiElement(DOUBLE_QUOTE_CLOSER)('"')
          PsiElement(R_SQUAREBRACKET)(']')
        PsiElement(ELVIS)('?:')
        UnaryExpression
          PsiElement(MINUS)('-')
          IntegerLiteral
            PsiElement(INTEGER)('1')
    PsiElement(SEMICOLON)(';')
    PsiElement(R_CURLYBRACKET)('}')